### ISSUE TITLE
OpenVPN and newer openssl do not work with PIA

### ISSUE THREAD

**Issue by blumoreno** [qdm12/gluetun#2376]
### Is this urgent?

No

### Host OS

Ubuntu 24.04 (As an VM in Proxmox)

### CPU arch

x86_64

### VPN service provider

Private Internet Access

### What are you using to run the container

docker-compose

### What is the version of Gluetun

Running version latest built on 2024-07-28T14:31:52.048Z (commit ddbfdc9)

### What's the problem 🤔

VPN no longer connect after making no config changes. I get "OpenSSL: error:068000E9:asn1 encoding routines::utctime is too short:" and "CRL: cannot read CRL from file", and "Host is unreachable".
The VPN then keeps restarting attempting to connect but never succeeds in doing so.
I have looked through the discussions already uploaded and tried their solutions when it seemed promising to no avail.
Thank you in advance for your time. 

I have tried:

- Using different server regions,
- Using different environment variables, 
- Changing PIA encryption preset to normal, 
- UFW disabled, 
- Deleting and repulluing image.
- Removing config files (only the servers.json file that I am aware of) and repulled image
- Restarting the Ubuntu node, 
- Changing DNS servers,
-  Running 'docker run --rm -v /home/*redacted*/docker/gluetun:/gluetun qmcgaw/gluetun format-servers "private internet access"'

### Share your logs (at least 10 lines)

```plain text
2024-07-28T23:12:08-05:00 INFO [routing] default route found: interface eth0, gateway 172.18.0.1, assigned IP 172.18.0.2 and family v4
2024-07-28T23:12:08-05:00 INFO [routing] local ethernet link found: eth0
2024-07-28T23:12:08-05:00 INFO [routing] local ipnet found: 172.18.0.0/16
2024-07-28T23:12:08-05:00 INFO [firewall] enabling...
2024-07-28T23:12:08-05:00 DEBUG [firewall] iptables --policy INPUT DROP
2024-07-28T23:12:08-05:00 DEBUG [firewall] iptables --policy OUTPUT DROP
2024-07-28T23:12:08-05:00 DEBUG [firewall] iptables --policy FORWARD DROP
2024-07-28T23:12:08-05:00 DEBUG [firewall] ip6tables --policy INPUT DROP
2024-07-28T23:12:08-05:00 DEBUG [firewall] ip6tables --policy OUTPUT DROP
2024-07-28T23:12:08-05:00 DEBUG [firewall] ip6tables --policy FORWARD DROP
2024-07-28T23:12:08-05:00 DEBUG [firewall] iptables --append INPUT -i lo -j ACCEPT
2024-07-28T23:12:08-05:00 DEBUG [firewall] ip6tables --append INPUT -i lo -j ACCEPT
2024-07-28T23:12:08-05:00 DEBUG [firewall] iptables --append OUTPUT -o lo -j ACCEPT
2024-07-28T23:12:08-05:00 DEBUG [firewall] ip6tables --append OUTPUT -o lo -j ACCEPT
2024-07-28T23:12:08-05:00 DEBUG [firewall] iptables --append OUTPUT -m conntrack --ctstate ESTABLISHED,RELATED -j ACCEPT
2024-07-28T23:12:08-05:00 DEBUG [firewall] ip6tables --append OUTPUT -m conntrack --ctstate ESTABLISHED,RELATED -j ACCEPT
2024-07-28T23:12:08-05:00 DEBUG [firewall] iptables --append INPUT -m conntrack --ctstate ESTABLISHED,RELATED -j ACCEPT
2024-07-28T23:12:08-05:00 DEBUG [firewall] ip6tables --append INPUT -m conntrack --ctstate ESTABLISHED,RELATED -j ACCEPT
2024-07-28T23:12:08-05:00 DEBUG [firewall] iptables --append OUTPUT -o eth0 -s 172.18.0.2 -d 172.18.0.0/16 -j ACCEPT
2024-07-28T23:12:08-05:00 DEBUG [firewall] ip6tables --append OUTPUT -o eth0 -d ff02::1:ff/104 -j ACCEPT
2024-07-28T23:12:08-05:00 DEBUG [firewall] iptables --append INPUT -i eth0 -d 172.18.0.0/16 -j ACCEPT
2024-07-28T23:12:08-05:00 INFO [firewall] enabled successfully
2024-07-28T23:12:08-05:00 INFO [storage] creating /gluetun/servers.json with 19425 hardcoded servers
2024-07-28T23:12:08-05:00 DEBUG [netlink] IPv6 is not supported after searching 1 routes
2024-07-28T23:12:08-05:00 INFO Alpine version: 3.20.2
2024-07-28T23:12:08-05:00 INFO OpenVPN 2.5 version: 2.5.10
2024-07-28T23:12:08-05:00 INFO OpenVPN 2.6 version: 2.6.11
2024-07-28T23:12:08-05:00 INFO Unbound version: 1.20.0
2024-07-28T23:12:08-05:00 INFO IPtables version: v1.8.10
2024-07-28T23:12:08-05:00 INFO Settings summary:
├── VPN settings:
|   ├── VPN provider settings:
|   |   ├── Name: private internet access
|   |   ├── Server selection settings:
|   |   |   ├── VPN type: openvpn
|   |   |   ├── Regions: CA Montreal, CA Toronto, CA Ontario
|   |   |   └── OpenVPN server selection settings:
|   |   |       ├── Protocol: UDP
|   |   |       └── Private Internet Access encryption preset: strong
|   |   └── Automatic port forwarding settings:
|   |       ├── Redirection listening port: disabled
|   |       ├── Use port forwarding code for current provider
|   |       ├── Forwarded port file path: /Media/forwarded_port.txt
|   |       └── Credentials:
|   |           ├── Username: *redacted*
|   |           └── Password: [set]
|   └── OpenVPN settings:
|       ├── OpenVPN version: 2.6
|       ├── User: [set]
|       ├── Password: [set]
|       ├── Private Internet Access encryption preset: strong
|       ├── Network interface: tun0
|       ├── Run OpenVPN as: root
|       └── Verbosity level: 1
├── DNS settings:
|   ├── Keep existing nameserver(s): no
|   ├── DNS server address to use: 127.0.0.1
|   └── DNS over TLS settings:
|       ├── Enabled: yes
|       ├── Update period: every 24h0m0s
|       ├── Unbound settings:
|       |   ├── Authoritative servers:
|       |   |   └── cloudflare
|       |   ├── Caching: yes
|       |   ├── IPv6: no
|       |   ├── Verbosity level: 1
|       |   ├── Verbosity details level: 0
|       |   ├── Validation log level: 0
|       |   ├── System user: root
|       |   └── Allowed networks:
|       |       ├── 0.0.0.0/0
|       |       └── ::/0
|       └── DNS filtering settings:
|           ├── Block malicious: yes
|           ├── Block ads: no
|           ├── Block surveillance: no
|           └── Blocked IP networks:
|               ├── 127.0.0.1/8
|               ├── 10.0.0.0/8
|               ├── 172.16.0.0/12
|               ├── 192.168.0.0/16
|               ├── 169.254.0.0/16
|               ├── ::1/128
|               ├── fc00::/7
|               ├── fe80::/10
|               ├── ::ffff:127.0.0.1/104
|               ├── ::ffff:10.0.0.0/104
|               ├── ::ffff:169.254.0.0/112
|               ├── ::ffff:172.16.0.0/108
|               └── ::ffff:192.168.0.0/112
├── Firewall settings:
|   └── Enabled: yes
├── Log settings:
|   └── Log level: debug
├── Health settings:
|   ├── Server listening address: 127.0.0.1:9999
|   ├── Target address: cloudflare.com:443
|   ├── Duration to wait after success: 5s
|   ├── Read header timeout: 100ms
|   ├── Read timeout: 500ms
|   └── VPN wait durations:
|       ├── Initial duration: 6s
|       └── Additional duration: 5s
├── Shadowsocks server settings:
|   └── Enabled: no
├── HTTP proxy settings:
|   └── Enabled: no
├── Control server settings:
|   ├── Listening address: :8000
|   └── Logging: yes
├── OS Alpine settings:
|   ├── Process UID: 1000
|   ├── Process GID: 1000
|   └── Timezone: America/Chicago
├── Public IP settings:
|   ├── Fetching: every 12h0m0s
|   ├── IP file path: /tmp/gluetun/ip
|   └── Public IP data API: ipinfo
├── Server data updater settings:
|   ├── Update period: 24h0m0s
|   ├── DNS address: 1.1.1.1:53
|   ├── Minimum ratio: 0.8
|   └── Providers to update: private internet access
└── Version settings:
    └── Enabled: yes
2024-07-28T23:12:08-05:00 INFO [routing] default route found: interface eth0, gateway 172.18.0.1, assigned IP 172.18.0.2 and family v4
2024-07-28T23:12:08-05:00 DEBUG [routing] ip rule add from 172.18.0.2/32 lookup 200 pref 100
2024-07-28T23:12:08-05:00 INFO [routing] adding route for 0.0.0.0/0
2024-07-28T23:12:08-05:00 DEBUG [routing] ip route replace 0.0.0.0/0 via 172.18.0.1 dev eth0 table 200
2024-07-28T23:12:08-05:00 INFO [firewall] setting allowed subnets...
2024-07-28T23:12:08-05:00 INFO [routing] default route found: interface eth0, gateway 172.18.0.1, assigned IP 172.18.0.2 and family v4
2024-07-28T23:12:08-05:00 DEBUG [routing] ip rule add to 172.18.0.0/16 lookup 254 pref 98
2024-07-28T23:12:08-05:00 INFO [dns] using plaintext DNS at address 1.1.1.1
2024-07-28T23:12:08-05:00 INFO [http server] http server listening on [::]:8000
2024-07-28T23:12:08-05:00 INFO [healthcheck] listening on 127.0.0.1:9999
2024-07-28T23:12:08-05:00 INFO [firewall] allowing VPN connection...
2024-07-28T23:12:08-05:00 DEBUG [firewall] iptables --append OUTPUT -d 66.115.142.58 -o eth0 -p udp -m udp --dport 1197 -j ACCEPT
2024-07-28T23:12:08-05:00 DEBUG [firewall] iptables --append OUTPUT -o tun0 -j ACCEPT
2024-07-28T23:12:08-05:00 DEBUG [firewall] ip6tables --append OUTPUT -o tun0 -j ACCEPT
2024-07-28T23:12:08-05:00 INFO [openvpn] OpenVPN 2.6.11 x86_64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD]
2024-07-28T23:12:08-05:00 INFO [openvpn] library versions: OpenSSL 3.3.1 4 Jun 2024, LZO 2.10
2024-07-28T23:12:08-05:00 INFO [openvpn] OpenSSL: error:068000E9:asn1 encoding routines::utctime is too short:
2024-07-28T23:12:08-05:00 INFO [openvpn] OpenSSL: error:0688010A:asn1 encoding routines::nested asn1 error:Field=revocationDate, Type=X509_REVOKED
2024-07-28T23:12:08-05:00 INFO [openvpn] OpenSSL: error:0688010A:asn1 encoding routines::nested asn1 error:Field=revoked, Type=X509_CRL_INFO
2024-07-28T23:12:08-05:00 INFO [openvpn] OpenSSL: error:0688010A:asn1 encoding routines::nested asn1 error:Field=crl, Type=X509_CRL
2024-07-28T23:12:08-05:00 INFO [openvpn] OpenSSL: error:0488000D:PEM routines::ASN1 lib:
2024-07-28T23:12:08-05:00 INFO [openvpn] CRL: cannot read CRL from file [[INLINE]]
2024-07-28T23:12:08-05:00 INFO [openvpn] CRL: loaded 0 CRLs from file -----BEGIN X509 CRL-----
2024-07-28T23:12:08-05:00 INFO [openvpn] MIIDWDCCAUAwDQYJKoZIhvcNAQENBQAwgegxCzAJBgNVBAYTAlVTMQswCQYDVQQIEwJDQTETMBEGA1UEBxMKTG9zQW5nZWxlczEgMB4GA1UEChMXUHJpdmF0ZSBJbnRlcm5ldCBBY2Nlc3MxIDAeBgNVBAsTF1ByaXZhdGUgSW50ZXJuZXQgQWNjZXNzMSAwHgYDVQQDExdQcml2YXRlIEludGVybmV0IEFjY2VzczEgMB4GA1UEKRMXUHJpdmF0ZSBJbnRlcm5ldCBBY2Nlc3MxLzAtBgkqhkiG9w0BCQEWIHNlY3VyZUBwcml2YXRlaW50ZXJuZXRhY2Nlc3MuY29tFw0xNjA3MDgxOTAwNDZaFw0zNjA3MDMxOTAwNDZaMCYwEQIBARcMMTYwNzA4MTkwMDQ2MBECAQYXDDE2MDcwODE5MDA0NjANBgkqhkiG9w0BAQ0FAAOCAgEAppFfEpGsasjB1QgJcosGpzbf2kfRhM84o2TlqY1ua+Gi5TMdKydA3LJcNTjlI9a0TYAJfeRX5IkpoglSUuHuJgXhP3nEvX10mjXDpcu/YvM8TdE5JV2+EGqZ80kFtBeOq94WcpiVKFTR4fO+VkOK9zwspFfb1cNs9rHvgJ1QMkRUF8PpLN6AkntHY0+6DnigtSaKqldqjKTDTv2OeH3nPoh80SGrt0oCOmYKfWTJGpggMGKvIdvU3vH9+EuILZKKIskt+1dwdfA5Bkz1GLmiQG7+9ZZBQUjBG9Dos4hfX/rwJ3eU8oUIm4WoTz9rb71SOEuUUjP5NPy9HNx2vx+cVvLsTF4ZDZaUztW9o9JmIURDtbeyqxuHN3prlPWB6aj73IIm2dsDQvs3XXwRIxs8NwLbJ6CyEuvEOVCskdM8rdADWx1J0lRNlOJ0Z8ieLLEmYAA834VN1SboB6wJIAPxQU3rcBhXqO9y8aa2oRMg8NxZ5gr+PnKVMqag1x0IxbIgLxtkXQvxXxQHEMSODzvcOfK/nBRBsqTj30P+R87sU8titOoxNeRnBDRNhdEy/QGAqGh62ShPpQUCJdnKRiRTjnil9hMQHevoSuFKeEMO30FQL7BZyo37GFU+q1WPCplVZgCP9hC8Rn5K2+f6KLFo5bhtowSmu+GY1yZtg+RTtsA=
2024-07-28T23:12:08-05:00 INFO [openvpn] -----END X509 CRL-----
2024-07-28T23:12:08-05:00 INFO [openvpn] TCP/UDP: Preserving recently used remote address: [AF_INET]66.115.142.58:1197
2024-07-28T23:12:08-05:00 INFO [openvpn] UDPv4 link local: (not bound)
2024-07-28T23:12:08-05:00 INFO [openvpn] UDPv4 link remote: [AF_INET]66.115.142.58:1197
2024-07-28T23:12:10-05:00 INFO [openvpn] read UDPv4 [EHOSTUNREACH|EHOSTUNREACH]: Host is unreachable (fd=3,code=113)
2024-07-28T23:12:14-05:00 INFO [openvpn] read UDPv4 [EHOSTUNREACH]: Host is unreachable (fd=3,code=113)
2024-07-28T23:12:14-05:00 INFO [healthcheck] program has been unhealthy for 6s: restarting VPN
2024-07-28T23:12:14-05:00 INFO [healthcheck] 👉 See https://github.com/qdm12/gluetun-wiki/blob/main/faq/healthcheck.md
2024-07-28T23:12:14-05:00 INFO [healthcheck] DO NOT OPEN AN ISSUE UNLESS YOU READ AND TRIED EACH POSSIBLE SOLUTION
2024-07-28T23:12:14-05:00 INFO [vpn] stopping
2024-07-28T23:12:14-05:00 INFO [vpn] starting
2024-07-28T23:12:14-05:00 INFO [firewall] allowing VPN connection...
2024-07-28T23:12:14-05:00 DEBUG [firewall] iptables --delete OUTPUT -d 66.115.142.58 -o eth0 -p udp -m udp --dport 1197 -j ACCEPT
2024-07-28T23:12:14-05:00 DEBUG [firewall] iptables --delete OUTPUT -o tun0 -j ACCEPT
2024-07-28T23:12:14-05:00 DEBUG [firewall] ip6tables --delete OUTPUT -o tun0 -j ACCEPT
2024-07-28T23:12:14-05:00 DEBUG [firewall] iptables --append OUTPUT -d 66.115.142.35 -o eth0 -p udp -m udp --dport 1197 -j ACCEPT
2024-07-28T23:12:14-05:00 DEBUG [firewall] iptables --append OUTPUT -o tun0 -j ACCEPT
2024-07-28T23:12:14-05:00 DEBUG [firewall] ip6tables --append OUTPUT -o tun0 -j ACCEPT
2024-07-28T23:12:14-05:00 INFO [openvpn] OpenVPN 2.6.11 x86_64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD]
2024-07-28T23:12:14-05:00 INFO [openvpn] library versions: OpenSSL 3.3.1 4 Jun 2024, LZO 2.10
2024-07-28T23:12:14-05:00 INFO [openvpn] OpenSSL: error:068000E9:asn1 encoding routines::utctime is too short:
2024-07-28T23:12:14-05:00 INFO [openvpn] OpenSSL: error:0688010A:asn1 encoding routines::nested asn1 error:Field=revocationDate, Type=X509_REVOKED
2024-07-28T23:12:14-05:00 INFO [openvpn] OpenSSL: error:0688010A:asn1 encoding routines::nested asn1 error:Field=revoked, Type=X509_CRL_INFO
2024-07-28T23:12:14-05:00 INFO [openvpn] OpenSSL: error:0688010A:asn1 encoding routines::nested asn1 error:Field=crl, Type=X509_CRL
2024-07-28T23:12:14-05:00 INFO [openvpn] OpenSSL: error:0488000D:PEM routines::ASN1 lib:
2024-07-28T23:12:14-05:00 INFO [openvpn] CRL: cannot read CRL from file [[INLINE]]
2024-07-28T23:12:14-05:00 INFO [openvpn] CRL: loaded 0 CRLs from file -----BEGIN X509 CRL-----
2024-07-28T23:12:14-05:00 INFO [openvpn]
```

### Share your configuration

```yml
version: "3"
services:
  gluetun:
    image: qmcgaw/gluetun
    container_name: gluetun
    cap_add:
      - NET_ADMIN
    devices:
      - /dev/net/tun:/dev/net/tun 
    ports:
      - 6881:6881
      - 6881:6881/udp
      - 8085:8085 # qbittorrent
      - 8989:8989 # Sonarr
      - 9696:9696 # Prowlarr
      - 7878:7878 # Radarr
      - 8191:8191 # FlareSolverr
    volumes:
      - /home/*redacted*/docker/gluetun:/gluetun 
      - /mnt/media:/Media 
    environment:
      - VPN_SERVICE_PROVIDER=private internet access  
      - OPENVPN_USER= # enter VPN username
      - OPENVPN_PASSWORD= # enter VPN password
      - SERVER_REGIONS=CA Montreal,CA Toronto,CA Ontario 
      - PORT_FORWARD_ONLY=true
      - VPN_PORT_FORWARDING=on
      - VPN_PORT_FORWARDING_STATUS_FILE=/Media/forwarded_port.txt 
      - TZ=America/Chicago
      - UPDATER_PERIOD=24h
    restart: unless-stopped
```

**Comment by blumoreno** [comment]
I just updated the title as I got the service running in v3.37.0 but still does not work on latest.

**Comment by hoonlight** [comment]
I have the same problem, it works fine with the PR-2253 image, but not with the latest image.

**Comment by ilmaio82** [comment]
Same issue here, i had to revert to V3

**Comment by illiteratealliterator** [comment]
Same issue. Works for me in v3.38.0 but not in latest.

**Comment by blumoreno** [comment]
All of these "same here"s are not going to be helpful for qdm12. Please explain your setup, docker-compose, logs/errors, and provider instead of just saying same here.

**Comment by qdm12** [comment]
This is likely due to the upgrade from Alpine 3.19 to 3.20 which also upgraded Openvpn. This seems related: https://github.com/openssl/openssl/discussions/24301

**Comment by ilmaio82** [comment]
I use wireguard and i have issue with latest release, my docker container keeps crash on boot.

**Comment by qdm12** [comment]
If anyone has the issue with a provider **OTHER** than Private Internet Access, please comment here!

@ilmaio82 this is about Openvpn+Pia, not Wireguard. You are not having the same issue. Please open another issue and avoid commenting low quality comments without even checking your logs.

**Comment by qdm12** [comment]
I pushed 264480b659976145d7133446ad15023605535053 which **removes** the `crl-verify` option, which may solve this, let me know. This should have no security influence, see the CRL notes in https://openvpn.net/community-resources/revoking-certificates/ if you wonder.

**Comment by ilmaio82** [comment]
> If anyone has the **SAME** issue with a provider **OTHER** than Private Internet Access, please comment here!
> 
> @ilmaio82 this is about Openvpn+Pia, not Wireguard. You are not having the same issue. Please open another issue and avoid commenting low quality comments without even checking your logs.

I was having the same issue, since it solved with your latest build https://github.com/qdm12/gluetun/commit/264480b659976145d7133446ad15023605535053

**Comment by qdm12** [comment]
Great, let's assume it's solved then. Closing this! If not, please open another issue 😉 Thanks!

**Comment by Iggloovortex** [comment]
Any tips on how to get this fix to work with docker? pulling the latest image still gives crl error.

**Comment by blumoreno** [comment]
> Any tips on how to get this fix to work with docker? pulling the latest image still gives crl error.

I had this happen to me momentarily. I deleted my config files and repulled the image.

**Comment by Iggloovortex** [comment]
For me I was using Custom vs PIA. Fix only applies when specifically using PIA. Once I switched over it and removed specific ports it worked. Thanks
